lib/string_reverse.rb

```diff
@@ -1,6 +1,18 @@
 # A method to reverse a string in place.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n is number of characters in the string
+# Space complexity: O(1)
 def string_reverse(my_string)
-  raise NotImplementedError
+  if my_string == nil
+    return nil
+  else
+    i = 0
+    j = my_string.length - 1
+    while i < j
+      temp = my_string[i]
+      my_string[i] = my_string[j]
+      my_string[j] = temp

```

Comment:
If you wanted, you could do this on one line:

```ruby
my_string[i], my_string[j] = my_string[j], my_string[i]
```

but it's just as valid to use a temporary variable.